pkg/tesla/fingerprint/fingerprint_test.go

```diff
@@ -0,0 +1,75 @@
+package fingerprint_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/DIMO-Network/model-garage/pkg/cloudevent"
+	"github.com/DIMO-Network/model-garage/pkg/tesla/fingerprint"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFullFromDataConversion(t *testing.T) {
+	t.Parallel()
+	event := cloudevent.CloudEvent[json.RawMessage]{}
+
+	for _, test := range []struct {
+		ExpectedVin   string
+		CloudEvent    string
+		ExpectedError error
```

Comment:
thought: The presence of this is kinda tantalizing. It would be nice to check the length thing, for example.